frontend/simple/views/Invite.vue

```diff
@@ -86,105 +33,108 @@
-          this.members.push({ state, contractId })
-        }
-        this.searchUser = null
-        this.userErrorMsg = ''
-      } catch (err) {
-        console.log(err)
-        this.userErrorMsg = L('Invalid User')
-      }
+  computed: {
+    isProposal () {
+      return this.memberCount() >= 3
     },
-    remove (index) {
-      this.members.splice(index, 1)
+    ...mapState([
+      'currentGroupId',
+      'loggedIn'
+    ]),
+    ...mapGetters([
+      'currentGroupState',
+      'memberCount'
+    ])
+  },
+  methods: {
+    updateInvitees (payload) {
+      this.form.errorMsg = null
+      Object.assign(this.form, payload.data)
     },
     async submit () {
       try {
-        this.errorMsg = null
-        // TODO: as members are successfully invited display in a
-        // seperate invitees grid and add them to some validation for duplicate invites
-        for (let member of this.members) {
-          // We need to have the latest mailbox attribute for the user
-          const mailbox = await backend.latestHash(member.state.attributes.mailbox)
-          const sentDate = new Date().toString()
+        this.form.errorMsg = null
+        const groupId = this.currentGroupId
+        const groupName = this.currentGroupState.groupName
+
+        for (let member of this.form.invitees) {
+          let mailbox = member.state.attributes.mailbox
+          let memberName = member.state.attributes.name
 
-          // We need to post the invite to the users' mailbox contract
-          const invite = new Events.HashableMailboxPostMessage(
-            {
-              from: this.$store.getters.currentGroupState.groupName,
-              headers: [this.$store.state.currentGroupId],
-              messageType: Events.HashableMailboxPostMessage.TypeInvite,
-              sentDate
-            },
-            mailbox
-          )
-          await backend.publishLogEntry(member.state.attributes.mailbox, invite)
+          if (this.isAlreadyInvited(member)) {
+            this.form.errorMsg = template(L('Failed to invite users: member {name} is already invited'),
+              { name: memberName })
+            return
+          }
 
-          // We need to make a record of the invitation in the group's contract
-          const latest = await backend.latestHash(this.$store.state.currentGroupId)
-          const invited = new Events.HashableGroupRecordInvitation(
-            {
-              username: member.state.attributes.name,
-              inviteHash: invite.toHash(),
-              sentDate
-            },
-            latest
-          )
-          await backend.publishLogEntry(this.$store.state.currentGroupId, invited)
+          let inviteToMailbox = await sbp('groupIncome.contracts/invite/mailRecord/create', mailbox, groupName, groupId)
+          let inviteToGroup = await sbp('groupIncome.contracts/invite/groupRecord/create', inviteToMailbox.toHash(), memberName, groupId)
```

Comment:
Nice, I like this direction 👍 

One thing though: I think it might make more sense to have the first thing after the domain be the contract, instead of an "abstract activity".

I.e. there is no `invite` contract.

So perhaps: `groupIncome.contracts/<contract>/<contract-action>`

Reply:
Sounds good, however, just make sure that `<contract-action>` is a 1-to-1 mapping with the classes that `extends HashableAction` in `shared/events.js`.

So there is no `createInvite` action ... however you could create another SBP selector to do something like that, but that would probably need to be done under a different selector path, not under the `groupIncome.contracts/<contract>/<contract-action>` style